homeassistant/components/media_player/harman_kardon_avr.py

```diff
@@ -0,0 +1,135 @@
+"""
+Support for interface with an Harman/Kardon or JBL AVR.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.harman_kardon_avr/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    PLATFORM_SCHEMA, SUPPORT_TURN_ON, SUPPORT_SELECT_SOURCE,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PORT)
+
+REQUIREMENTS = ['hkavr==0.0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Harman Kardon AVR'
+DEFAULT_PORT = 10025
+
+SUPPORT_HARMAN_KARDON_AVR = SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | \
+                            SUPPORT_TURN_OFF | SUPPORT_TURN_ON | \
+                            SUPPORT_SELECT_SOURCE
+
+SOURCES = ["Disc", "STB", "Cable Sat", "Media Server", "DVR", "Radio", "TV",
+           "USB", "Game", "Home Network", "AUX", "AVR"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+def setup_platform(hass, config, add_devices, discover_info=None):
+    """Set up the AVR platform."""
+    import hkavr
+
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    avr = hkavr.HkAVR(host, port, name)
+
+    avr_device = HkAvrDevice(avr)
+    if avr_device is None:
+        _LOGGER.warning("Could not connect to AVR")
+        return
+
+    add_devices([avr_device], True)
+
+
+class HkAvrDevice(MediaPlayerDevice):
+    """Representation of a Harman Kardon AVR / JBL AVR TV."""
+
+    def __init__(self, avr):
+        """Initialize a new HarmanKardonAVR."""
+        self._avr = avr
+
+        self._name = avr.name
+        self._host = avr.host
+        self._port = avr.port
+
+        self._source_list = SOURCES
```

Comment:
If we will use the constant, we don't need the instance attribute. We can use the constant directly.